Da-Bois-Tracking/Tracking-App
# TA-15 Dockerize Project

### In this PR, I'm adding/changing the following:

- Implementing Docker
- Containerizing the Frontend, backend, and the database. 

As we continue development, Dockerizing it will help streamline the development process and ensure consistency across different development environments. Docker containers provide a lightweight and portable solution for packaging our application and its dependencies.

### Please do the following to test my code:

1. Download Docker:
`https://www.docker.com/products/docker-desktop/`
After you finish downloading Docker, make sure to run it. 

2.  Checkout to my feature branch
   `git checkout TA-15-Dockerize-Project`

3. Create a .env file in the following places:
   - root directory
   - Backend folder

4. Copy the environment variables into your `.env` files (located in the `Discord - Backend` channel.

5. Navigate to the Backend folder:
   `cd backend`
   - Start your Virtual Environment
      - For Windows:
         `venv/Scripts/activate`
      - For Mac:
         `source venv/bin/activate`
   
6. Install the backend dependencies
   `pip install -r requirements.txt`

7. Navigate to the Frontend folder:
   `cd frontend`
   `npm install`

8. In your terminal, run:
   `docker compose build --no-cache`
   `docker compose up`

9.  TEST.
   - Navigate to `localhost:3000`. Here, you should see the frontend built in Vite.
   - Navigate to `localhost:8000`. Here, you should see a JSON body with Hello World.
   - Navigate to `localhost:8000/docs`. Here you should see example endpoints. Test them out. 

[Related Jira Ticket](https://bois-tracking-app.atlassian.net/browse/TA-15)
